### PR TITLE
Make uploading to geoserver identical for both shapefile and geopackage

### PR DESCRIPTION
**Pull request by ywkim312**
Please test with the attached zip file. 
After unzip it, you will see one geopackage file (guid_test.gpkg) and others that are shapefile component (epn_guid.shp)

[data.zip](https://github.com/IN-CORE/incore-services/files/14104257/data.zip)

Test posting both dataset
 
Testing GeoPackage
1. create a dataset with 
{"format": "geopackage", "title": "test-delete", "description": "delete me", "dataType": "incore:addressPoints"} 
please ignore dataType, this is just a test
2. attach geopackage file to the dataset created in step 1
3. if attach process is successful, the PR is good
4. remove the dataset created in step 1

Testing Shapefile
1. create a dataset with
{"format": "shapefile", "title": "test-delete", "description": "delete me", "dataType": "incore:addressPoints"} 
please ignore dataType, this is just a test
2. attach shpefile components (.shp, .dbf, .shx, .prj) to the dataset created in step 1
3. if attach process is successful, the PR is good
4. remove the dataset created in step 1

### CODE REVIEW

**Review by ylyangtw — APPROVED** [review]
uploaded both shapefile and geopackage successfully. change look good

**Comment by navarroc** [comment]
Can you add a copyright header to the GeoserverRestApi class?

File: server/data-service/src/main/java/edu/illinois/ncsa/incore/service/data/utils/GeoserverRestApi.java [hunk 1]
```diff
@@ -140,51 +140,78 @@ public boolean postFileToGeoserver(String apiUrl, File inData, String contType)
...
+        try {
+            String restUrl = this.geoserverUrl + "/rest";
+            String fileNameNoExt = fileName.split("\\.")[0];
+            int datastoreResponse = createDatastore(restUrl, GEOSERVER_WORKSPACE, store, inFile.getAbsolutePath(), fileForamt);
+            int layerResponse = createLayer(restUrl, GEOSERVER_WORKSPACE, store, store, fileNameNoExt);
+            if (datastoreResponse == 201 && layerResponse == 201) {
+                published = true;
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```
Comment (navarroc) [inline server/data-service/src/main/java/edu/illinois/ncsa/incore/service/data/utils/GeoserverRestApi.java:209]: Maybe add a logger.error here so we capture the log?

**Comment by ywkim312** [comment]
> Can you add a copyright header to the GeoserverRestApi class?

Added

File: server/data-service/src/main/java/edu/illinois/ncsa/incore/service/data/utils/GeoserverRestApi.java (hunk 1 above)
Comment (ywkim312) [reply server/data-service/src/main/java/edu/illinois/ncsa/incore/service/data/utils/GeoserverRestApi.java:219]: added logger.error

**Comment by navarroc** [comment]
How do I test this? Can I test it on my local or do I need to use port forwarding to test on dev?

**Comment by ywkim312** [comment]
> How do I test this? Can I test it on my local or do I need to use port forwarding to test on dev?

You can test it in local. Maybe let me deploy this to dev now and will let you know

**Comment by navarroc** [comment]
> > How do I test this? Can I test it on my local or do I need to use port forwarding to test on dev?
> 
> You can test it in local. Maybe let me deploy this to dev now and will let you know

I was able to test on my local. No need to deploy on dev.

**Comment by navarroc** [comment]
Can you remove the system.out print statements in GeoserverRestAPI?

**Comment by ywkim312** [comment]
> Can you remove the system.out print statements in GeoserverRestAPI?

removed

**Review by navarroc — APPROVED** [review]

**Pull request merged**